java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -0,0 +1,92 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.NonNull;
+
+public class WeightAggregateOptions {
+    public static final String WEIGHTS_REDIS_API = "WEIGHTS";
+    public static final String AGGREGATE_REDIS_API = "AGGREGATE";
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    public enum Aggregate {
+        /** Aggregates by summing the scores of each element across sets. */
+        SUM,
+        /** Aggregates by selecting the minimum score for each element across sets. */
+        MIN,
+        /** Aggregates by selecting the maximum score for each element across sets. */
+        MAX;
+    }
+
+    /**
+     * Basic interface. Please use one of the following implementations:
+     *
+     * <ul>
+     *   <li>{@link KeyArray}
+     *   <li>{@link WeightedKeys}
+     * </ul>
+     */
+    public interface WeightableKeys {
```

Comment:
The interface name should describe its capabilities (for this: can be convertible toArgs(), or to argument tokens) - So maybe `KeysTokenizer` or `KeysToArgsTokenizer` or simply `KeysToArgs`.  

Since the purpose of this interface is also to union Keys and WeightedKeys, it could also be something simple like `KeysWithOptionalWeight` or `KeysOrWeightedKeys`.  

I prefer the last option `KeysOrWeightedKeys` since it's straightforward and simple for our interface.